processor/src/main/java/org/mapstruct/ap/internal/util/Message.java

```diff
@@ -53,6 +54,8 @@
     PROPERTYMAPPING_INVALID_PROPERTY_NAME( "No property named \"%s\" exists in source parameter(s)." ),
     PROPERTYMAPPING_NO_PRESENCE_CHECKER_FOR_SOURCE_TYPE( "Using custom source value presence checking strategy, but no presence checker found for %s in source type." ),
     PROPERTYMAPPING_NO_READ_ACCESSOR_FOR_TARGET_TYPE( "No read accessor found for property \"%s\" in target type." ),
+    PROPERTYMAPPING_POSSIBLE_NULL_VALUE_WHEN_NO_WRITE_ACCESSOR_TYPE( "NullPointer risk when only a read accessor is available and applying NullValueMappingStrategy.RETURN_NULL for property \"%s\" in target type.", Diagnostic.Kind.WARNING ),
+    PROPERTYMAPPING_POSSIBLE_NULL_VALUE_UPDATE_METHOD_ACCESSOR_TYPE( "NullPointer risk when using update method and applying NullValueMappingStrategy.RETURN_NULL for property \"%s\" in target type.", Diagnostic.Kind.WARNING ),
```

Comment:
There are two new test cases in the issue #289 reproducer. I will move them (in due time) to collection testing. Do they answer your question? I can of course externalize them..